kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -121,4 +125,33 @@ public Path getLogPath() {
     public Path getDataPath() {
         return dataPath;
     }
+
+    /**
+     * Returns the timestamp of the latest commit of this snapshot.
+     * For an uninitialized snapshot, this returns -1.
+     * <p>
+     * When InCommitTimestampTableFeature is enabled, the timestamp
+     * is retrieved from the CommitInfo of the latest commit which
+     * can result in an IO operation.
```

Comment:
Please also mention that for non-ICT tables, this is the same as the file modification time of the latest commit in the snapshot.